zap/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+mod irgen;
+mod output;
+mod parser;
+mod util;
+
+use thiserror::Error;
+use wasm_bindgen::prelude::*;
+
+#[derive(Error, Debug)]
+pub enum Error {
+	#[error("Config parser error: {0}")]
+	ParseError(String),
+	#[error("File System error: {0}")]
+	FSError(#[from] std::io::Error),
+	#[error("Unknown type referenced: `{0}`")]
+	UnknownTypeReference(String),
+	#[error("Duplicate type referenced: `{0}`")]
+	DuplicateType(String),
+	#[error("Duplicate Event referenced: `{0}`")]
+	DuplicateEvent(String),
+}
```

Comment:
I wasn't sure about my labelling of the errors. please confirm if you are happy with these (they will appear in the playground etc)

Reply:
change the name of `UnknownTypeReference` to `UnknownTypeRef`. and for duplicate types and events change the error to `Duplicate type/event declared` instead of referenced.